### DISCUSSION TITLE
Edit call back URL

### DISCUSSION THREAD

**Discussion by KirstyPeace** [etsy/open-api#882] [Q&A]
How do I edit the “call back url” that I need in order to refresh access?

**Comment by benjymouse** [comment]
Go here: https://www.etsy.com/developers/your-apps

You will see a link called "Edit your callback URLs"

**Comment by KirstyPeace** [comment]
I can't see this option, attached is what i see?
![image](https://user-images.githubusercontent.com/93130554/221203293-95901c26-fb0d-408e-b904-be0aec30421c.png)

**Comment by Gareth064** [comment]
Click on the name of your app
![image](https://user-images.githubusercontent.com/22307223/221220262-3a8134cf-4c73-48ff-9465-f12e345b29b9.png)

**Comment by KirstyPeace** [comment]
![image](https://user-images.githubusercontent.com/93130554/221224259-fe0fcf01-9452-4251-9e95-3a84ed2a3c33.png)
Attached is all I can see?

**Reply by Gareth064** [reply]
It's possibly missing then because they have't yet approved you're app for us

![image](https://user-images.githubusercontent.com/22307223/221225798-f02fc503-b0c3-4cef-ad82-864e53005824.png)

**Reply by Gareth064** [reply]
Email developer@etsy.com to ask them what's going on.

**Reply by Gareth064** [reply]
You'd see this when its approved I suppose

![image](https://user-images.githubusercontent.com/22307223/221226533-3de9ca08-cb2d-46e3-adda-519874da7ace.png)

**Reply by Gareth064** [reply]
@KirstyPeace did your app ever get approved, and the callback URL button appear?

**Reply by ruslankiskinov** [reply]
The app was created by my employer at least 6 years ago. There are internal apps that are using Etsy API V2 for many years. I need to migrate them.
The internal apps already have ConsumerKey, ConsumerSecret, RequestToken and RequestTokenSecret pairs which are used now.

Here is the screenshot of the dev account they give me. 
![EtsyApp](https://user-images.githubusercontent.com/7082935/223198079-a4c9d33a-05f9-42e4-9ec1-0bfbdbe2c12f.png)

It states Commercial Access although API is used only for internal needs.
When you click on the app name, nothing is opening the click does nothing.

**Reply by Gareth064** [reply]
If it's an internal app then what's stopping you from creating a new App?

Most developers would have struggled with creating a new app because, users (external) would have had to reauthorize their app with the stores again and it would have been a nightmare

If it's an internal/personal app where no other Etsy shop owners use it, then you'd be better off creating a new app.

But if you're determined to keep the same app then you'd need to email the Etsy Dev team for help. They don't monitor this GitHub anymore to offer support. developer@etsy.com

**Reply by ruslankiskinov** [reply]
I created a new app a couple of days ago and I'm waiting for approval. Why do I need to wait for approval anyway? How much do I have to wait?

**Comment by ruslankiskinov** [comment]
I want to migrate my old application to V3 API but this callback URL does not exist for the legacy apps!
How can I migrate the old app? There is no way to enter the callback URL. When I click on the name of the app nothing happens.

**Reply by Gareth064** [reply]
When you say "nothing happens" what does that mean? Can you show a screenshot

**Comment by Gareth064** [comment]
The million dollar question that no-one here can answer.

You can reach out to the Etsy Dev team via email to chase it.

***@***.***



Kind Regards

Gareth Doherty



It’s nice to be important,

But it’s more important to be nice.



________________________________
